packages/web/src/common/store/pages/audio-rewards/selectors.ts

```diff
@@ -7,9 +9,24 @@ export const getTrendingRewardsModalType = (state: CommonState) =>
 export const getChallengeRewardsModalType = (state: CommonState) =>
   state.pages.audioRewards.challengeRewardsModalType
 
-export const getUserChallenges = (state: CommonState) =>
+export const getUserChallengeSpecifierMap = (state: CommonState) =>
   state.pages.audioRewards.userChallenges
 
+// Returns just a single challenge per challengeId
+export const getUserChallenges = createSelector(
```

Comment:
Should this even exist? Seems like it could lead to confusion or bugs in the future. If we're keeping it, what if it returns the _list_ of challenges for that ID and let the consumer decide how to prioritize? Also curious if we should be more sophisticated than taking the arbitrary first challenge?

Reply:
Yeah... in my initial stab at this I rewrote this selector return multiple challenges per challenge_id, but it ends up being quite the sweeping code change to have to change the call site every single time we call this selector. Most of the time we call this wanting to know some property about the challenge itself (like the amount, the type, etc), which is the same for every instance of the challenge.

On top of all of this, it really feels like trending challenges are an edge case - it's the only challenge you can complete multiple times, since aggregate challenges get rolled up in the response from DN. Down to revisit this if the state modeling doesn't take us where we want it to go, but don't want to spend too much time for this edge case

Reply:
Makes sense - maybe we can defensively remove specifier specific info from this selector at some point in the future instead and make it clear not to use this for those cases?